### ISSUE TITLE
Progress API used wrongly by LSPIDEServices?

### ISSUE THREAD

**Issue by jurgenvinju** [usethesource/rascal-language-servers#115]

**Comment by rodinaarssen** [comment]
Does the output view give any hints as to which request?

**Comment by DavyLandman** [comment]
which version of vs code are you running?

What is the command you are running inside the REPL?

**Comment by jurgenvinju** [comment]
I am running VScode 1.63.0, usethesource.rascalmpl v0.1.5

I'm am loading the modules of the compiler in rascal-core, via `import lang::rascalcore::compile::Compile;`

**Comment by rodinaarssen** [comment]
When opening that file in an editor, I saw these messages in the Output:
```
2021-12-16 10:46:27,562 [pool-2-thread-1] ERROR - org.rascalmpl.vscode.lsp.rascal.RascalTextDocumentService Error opening file |project://typepal/src/analysis/typepal/Solver.rsc| to get contents
org.rascalmpl.uri.UnsupportedSchemeException: Unsupported scheme project
```
Weird error, but I think `typepal` may be unresolvable? Adding that project to my workspace solved it for me

**Comment by jurgenvinju** [comment]
The error is weird. That does happen because the project scheme is registered _per_ authority (so for every project separately). 

So someone needs to refer to `project://typepal` for some reason, but that is a bug in itself. The typepal project is only reachable through `lib://typepal`

**Comment by DavyLandman** [comment]
I think this comes from the rascal-core checker. when reporting errors. maybe since it get's the source locations from the tpl?

**Comment by rodinaarssen** [comment]
These popups appear at import time, though

**Comment by jurgenvinju** [comment]
I think I am  using the progress API in the wrong way somewhere. Before those errors would be eaten/silent. That has improved and now they are not silent anymore. An easy test case is: start a terminal, type `#int` because that will trigger loading of the parser generator which uses the IRascalMonitor to report progress. That monitor is wrapped to report to the progress API here:

https://github.com/usethesource/rascal-language-servers/blob/a5a3f619416036afd0bfd498cb697963feae5c64/rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/EvaluatorUtil.java#L106

**Comment by jurgenvinju** [comment]
https://github.com/usethesource/rascal-language-servers/blob/a5a3f619416036afd0bfd498cb697963feae5c64/rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/LSPIDEServices.java#L71

**Comment by jurgenvinju** [comment]
The bug must be somewhere here:

```
@Override
    public void jobStart(String name, int workShare, int totalWork) {
        String id = jobId(name);
        if (!jobs.contains(id)) {
            languageClient.createProgress(new WorkDoneProgressCreateParams(Either.forLeft(id)));
            jobs.add(id);
        }
        else {
            logger.warn("Double registration of job ignored: " +  id);
        }
    }

    private String jobId(String name) {
        // every job must be unique but several Threads may be producing these
        // kinds of jobs. So we add a unique number of spaces to every name
        // per object instance of this class.
        return myInstance + ":" + name;
    }

    @Override
    public void jobStep(String name, String message, int workShare) {
        if (jobs.contains(name)) {
            languageClient.notifyProgress(
                new ProgressParams(
                        Either.forLeft(jobId(name)),
                        Either.forLeft(new WorkDoneProgressReport()))
                    );
        }
        else {
            logger.warn("stepping a job that was not started: " + name);
        }
    }

    @Override
    public int jobEnd(String name, boolean succeeded) {
        String id = jobId(name);

        if (jobs.contains(id)) {
            jobs.remove(id);
            languageClient.notifyProgress(
                new ProgressParams(
                    Either.forLeft(id),
                    Either.forLeft(new WorkDoneProgressEnd())
                ));
            return 1;
        }
        else {
            logger.warn("ended a job that does not exist:" + name);
            return 1;
        }
    }
```

**Comment by DavyLandman** [comment]
Fixed in 79512236f89c8a4609f05435bcfad37716ab4dc2 and 8b822b4f15917a077941707dcf98a85ce2535806
